Python3 migration

sagepe opened the pull request:
Some small fixes to code via `2to3` and related bits to enable a deploy using Python 3 and Django 3.2. We'll look at the question of the development environment separately.

sagepe committed:
Additional adjustments for full Python3 and Django 3.2 support

This commit has a few minor changes for more complete support.

sagepe committed:
Exclude `.python-version` from Git

sagepe committed:
Update packages and setup scripts for Python3 and Django 3.2

This updates Django to 3.2.13. In addition, it moves several system
packages into the requirements file and setup scripts and no longer sets
`--system-site-packages`, hence the separate installation of the `wheel`
package.

Note that the GDAL install isn't included in the requirements file
as this must match the version of GDAL on the system, hence the
version check in the setup script; assuming libgdal-dev is installed
and the version matches, the `pip install` should work on recent
versions without setting `C_INCLUDE_PATH` and `CPLUS_INCLUDE_PATH`.

The versions used for the other dependencies are based on the versions
distributed with Debian Bullseye.

dracos commented on conf/packages:
```diff
@@ -1,14 +1,13 @@
-libgdal1 | libgdal20
+libgdal1 | libgdal20 | libgdal28
```
Do you want to change this to libgdal-dev and drop the python3- packages as we're doing in https://github.com/mysociety/mapit/pull/388? Only query now I think about that is we dropped python3-memcache and python3-gdal from that PR, but it looks like the code would still use them if present, so not sure if that was a mistake.

sagepe replied:
I've made a couple of changes to address this, hopefully.

dracos reviewed (commented):
Looks fine, one query about the packages file.

sagepe committed:
Permit configuration of `DEFAULT_AUTO_FIELD`

From Django 3.2, Warnings are generated if `DEFAULT_AUTO_FIELD` isn't
specified as the default value, `django.db.models.AutoField`, may
change in future and generate potentially unwanted migrations.

This commit sets the default to `django.db.models.AutoField` and adds
support for configuring this value in `general.yml`.

New projects may wish to set this to a different value, see the Django
documentation for more information.
